### DISCUSSION TITLE
Error on Build

### DISCUSSION THREAD

**Discussion by mikecasas** [oqtane/oqtane.framework#681] [General] [answered]
Is anyone else getting this? I download oqtane from the repo, then in a command line navigate to the Oqtane.Server folder. I run 
```
dotnet build -c Release
```
no problems. It builds as normal, then I run 
```
dotnet publish -c Release
```
and I get these errors..

![error1](https://user-images.githubusercontent.com/11261164/90188514-93e8ff80-dd89-11ea-981a-0b953c4f9d53.PNG)

I guess it's looking for the Oqtane.Upgrade.dll and other Upgrade assets.

**Comment by jimspillane** [comment, marked as answer]
Hey @mikecasas I have the same behavior.  I had to jump over to the _\oqtane.framework\Oqtane.Upgrade_ folder and run 
`dotnet build -c Release` first to copy the files.

**Reply by mikecasas** [reply]
Thanks @jimspillane. I will look into building that project first. I had a build pipeline, and those changes broke it. Any thoughts on why you have to build it first? Should the Upgrade project be moved up in the build order?

**Reply by jimspillane** [reply]
Adding a ProjectReference for the Upgrade project to the Oqtane.Server.csproj works. See if that helps.

```
  <ItemGroup>
    <ProjectReference Include="..\Oqtane.Client\Oqtane.Client.csproj" />
    <ProjectReference Include="..\Oqtane.Shared\Oqtane.Shared.csproj" />
    <ProjectReference Include="..\Oqtane.Upgrade\Oqtane.Upgrade.csproj" />
  </ItemGroup>
```

**Reply by mikecasas** [reply]
Thanks! Yes adding the reference fixed the publish. I will submit a PR.
